src/include/ceph_assert.h

```diff
@@ -144,4 +144,10 @@ using namespace ceph;
    ? _CEPH_ASSERT_VOID_CAST (0)					\
    : ::ceph::__ceph_assertf_fail (__STRING(expr), __FILE__, __LINE__, __CEPH_ASSERT_FUNCTION, __VA_ARGS__))
 
+#define consteval_assert(expr, msg)	\
```

Comment:
Is the idea here to stop compilation by throwing from a consteval context? Why not use static_assert? (I haven't seen this code before, I see lots happens in macros, there maybe very good reasons for it.)

Reply:
https://stackoverflow.com/a/57226813